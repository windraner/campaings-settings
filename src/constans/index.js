export const MENU_LIST = [
  'Campaign details',
  'Media Collections',
  'Blacklists',
  'Media rights'
];

export const ACTIVE_TAB = 'activeTab';
export const NAME = 'name';
export const DESCRIPTION = 'description';
export const PRODUCT_CATALOGUE = 'productCatalogue';
export const IS_SHOPPABLE_LINK = 'isShoppableLink';
export const IS_MEDIA_COLLECTION = 'isMediaCollection';
export const MEDIA_COLLECTION_LIST = 'mediaCollectionList';
export const IS_PHOTO_MODERATION_QUEUE = 'isPhotosModerationQueue';
export const USER_BLACK_LIST = 'userBlackList';
export const HASHTAG_BLACK_LIST = 'hashtagBlackList';
export const ACCOUNTS_SETTINGS_ITEMS = 'accountsSettingsItems';
export const SHOW_ERROR_ALERT = 'showErrorAlert';
export const SHOW_SUCCESS_ALERT = 'showSuccessAlert';
export const IS_MEDIA_RIGHTS = 'isMediaRights';
export const HASHTAG_RIGHTS_LIST = 'hashtagRightsList';
export const TERMS_AND_CONDITIONS = 'termsAndConditions';

export const VALIDATED_FIELDS = [
  ACTIVE_TAB
];
