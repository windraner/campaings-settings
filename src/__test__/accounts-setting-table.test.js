import React from 'react';
import AccountsSettingTable from '../components/main-content-section/tab-block/common/accounts-setting-table/accounts-setting-table';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import image from '../test.png';
import toJson from 'enzyme-to-json';

import { ACCOUNTS_SETTINGS_ITEMS } from '../constans';

configure({ adapter: new Adapter() });

const mock = {
  accountsSettingsItems: [
    {
      id: '1',
      url: image,
      userName : 'Instagram',
      platform: 'Instagram',
      status: 'Verified',
      userFeed: true,
      taggedFeed: true,
      mentionedFeed: false
    },
    {
      id: '2',
      url: image,
      userName : 'Blackmirror',
      platform: 'Twitter',
      status: 'Rejected',
      userFeed: false,
      taggedFeed: false,
      mentionedFeed: false
    }
  ],
  stateHandler() {}
};

const result = [
  {
    id: '1',
    url: image,
    userName : 'Instagram',
    platform: 'Instagram',
    status: 'Verified',
    userFeed: true,
    taggedFeed: true,
    mentionedFeed: true
  },
  {
    id: '2',
    url: image,
    userName : 'Blackmirror',
    platform: 'Twitter',
    status: 'Rejected',
    userFeed: false,
    taggedFeed: false,
    mentionedFeed: false
  }
];

describe('<AccountsSettingTable />', () => {
  it('Check currentSwitchHandler', () => {
    const onClickFunc = jest.fn();
    const wrapper = shallow(<AccountsSettingTable {...mock} stateHandler={onClickFunc} />);
    wrapper.instance().currentSwitchHandler({id: '1', field: 'mentionedFeed'});
    expect(onClickFunc).toBeCalled();
    expect(onClickFunc.mock.calls).toEqual([[ACCOUNTS_SETTINGS_ITEMS, result]]);
  });
  it('matches the snapshot hide', () => {
    const tree = shallow(<AccountsSettingTable {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
