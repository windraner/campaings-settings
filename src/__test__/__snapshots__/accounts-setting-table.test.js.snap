// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountsSettingTable /> matches the snapshot hide 1`] = `
<div
  className="campaigns-settings__accounts-setting-table-container"
>
  <div
    className="campaigns-settings__accounts-setting-table-header-container d-grid"
  >
    <div>
      Username / Platform
    </div>
    <div>
      Status
    </div>
    <div>
      <span
        className="position-relative"
      >
        User feed
        <TooltipIcon
          text="1111"
        />
      </span>
    </div>
    <div>
      <span
        className="position-relative"
      >
        Tagged feed
        <TooltipIcon
          text="2222"
        />
      </span>
    </div>
    <div>
      <span
        className="position-relative"
      >
        Mentioned feed
        <TooltipIcon
          text="3333"
        />
      </span>
    </div>
  </div>
  <AccountsSettingItem
    currentSwitchHandler={[Function]}
    item={
      Object {
        "id": "1",
        "mentionedFeed": false,
        "platform": "Instagram",
        "status": "Verified",
        "taggedFeed": true,
        "url": "test.png",
        "userFeed": true,
        "userName": "Instagram",
      }
    }
    key="1"
  />
  <AccountsSettingItem
    currentSwitchHandler={[Function]}
    item={
      Object {
        "id": "2",
        "mentionedFeed": false,
        "platform": "Twitter",
        "status": "Rejected",
        "taggedFeed": false,
        "url": "test.png",
        "userFeed": false,
        "userName": "Blackmirror",
      }
    }
    key="2"
  />
</div>
`;
